cmd/notary-signer/passwordwrap/interface.go

```diff
@@ -0,0 +1,19 @@
+package passwordwrap
+
+// Storage is the interface to store/retrieve password from any specific implementation type.
+type Storage interface {
+	// Get password from storage.
+	GetPassword(alias string) (string, error)
+
+	// Set password in storage.
+	SetPassword(alias string, newPassword string) error
+}
+
+// Protector is the interface to wrap/unwrap password using any specific implementation type.
+type Protector interface {
```

Comment:
Now that I'm seeing an implementation, I'm wondering if it makes sense for a `Protector` to really just be a wrapping struct that also implements the `Storage` interface. Am I missing something that would make that impossible?

i.e. (over simplified example just for demonstration, also only implemented Get side, hopefully Set can be inferred):
```
type PassStore struct {
    passwds map[string]string
}

func (ps PassStore) Get(alias) (string,error) {
    if p, ok := ps.passwds[alias]; ok {
        return p, nil
    }
    return "", fmt.Errorf("password not found for alias: %s", alias)
}

type Protector struct {
    store Storage
}

func NewProtector(s Storage) Storage {
   return &Protector{store: s}
}

func (p Protector) Get(alias) (string, error) {
    pass, err := p.store.Get(alias) 
    if err != nil {
        return "", err
    }
    return p.decrypt(pass) // where func (p Protector) decrypt(pass string) (string, error)
}
```
This will simplify the dependency injection later by wrapping up the storage and protector in to a single object that needs to be injected.